pkg/splunk/enterprise/configuration.go

```diff
@@ -999,6 +1000,24 @@ func updateSplunkPodTemplateWithConfig(ctx context.Context, client splcommon.Con
 	// append any extra variables
 	env = append(env, extraEnv...)
 
+	// check if there are any duplicate entries
+	// we use orderedmap so the test case can pass as json marshal
+	// expects order
+	if len(env) > 0 {
```

Comment:
Can we optimize to a single loop? Pseudo code below. We can add to ordered map?
```
func removeDuplicate[T string | int](sliceList []T) []T {
    allKeys := make(map[T]bool)
    list := []T{}
    for _, item := range sliceList {
        if _, value := allKeys[item]; !value {
            allKeys[item] = true
            list = append(list, item)
        }
    }
    return list
}
```

Reply:
We have not used generics anywhere in the code, I am not sure if this is the place to start. or may be its ok. will try adding this